gcal.py

```diff
@@ -0,0 +1,100 @@
+import datetime
+from datetime import date, timedelta
+import os.path
+import json
+
+from google.auth.transport.requests import Request
+from google.oauth2.credentials import Credentials
+from google_auth_oauthlib.flow import InstalledAppFlow
+from googleapiclient.discovery import build
+from googleapiclient.errors import HttpError
+
+# If modifying these scopes, delete the file token.json.
+SCOPES = ["https://www.googleapis.com/auth/calendar.events"]
+
+
+def main():
+  """Shows basic usage of the Google Calendar API.
+  Prints the start and name of the next event from each calendar
+  of a user in which "DPS" appears in the calendar name.
+  """
+  creds = None
+  # The file token.json stores the user's access and refresh tokens, and is
+  # created automatically when the authorization flow completes for the first
+  # time.
+  if os.path.exists("token.json"):
+    creds = Credentials.from_authorized_user_file("token.json", SCOPES)
+  # If there are no (valid) credentials available, let the user log in.
+  if not creds or not creds.valid:
+    if creds and creds.expired and creds.refresh_token:
+      creds.refresh(Request())
+    else:
+      flow = InstalledAppFlow.from_client_secrets_file(
+        "credentials.json", SCOPES
+      )
+      creds = flow.run_local_server(
+        host='localhost',
+        port=8088,
+        authorization_prompt_message='Please visit this URL: {url}',
+        success_message='The auth flow is complete; you may close this window.',
+        open_browser=True)

```

Comment:
I suspect that this flow will not work when automated via GitHub Actions. At some point we may want to separate the bits that can be automated and those that require manual intervention.

Reply:
If there is a mechanism for secrets or credentials for Actions to use, we may be ablet to add the token and credentials to the repo settings?